llvm/lib/IRReader/IRReader.cpp

```diff
@@ -91,6 +93,37 @@ std::unique_ptr<Module> llvm::parseIR(MemoryBufferRef Buffer, SMDiagnostic &Err,
                            [](StringRef, StringRef) { return std::nullopt; }));
 }
 
+std::unique_ptr<Module> llvm::parseIR(MemoryBufferRef Buffer, SMDiagnostic &Err,
+                                      LLVMContext &Context,
+                                      ParserCallbacks Callbacks) {
+  llvm::ArrayRef<uint8_t> BufferRef(
+      reinterpret_cast<const uint8_t *>(Buffer.getBufferStart()),
+      reinterpret_cast<const uint8_t *>(Buffer.getBufferEnd()));
```

Comment:
Why not using `StringRef`?

Reply:
What do you mean?
`llvm::parseIR` has a fixed prototype which takes `MemoryBufferRef`
`compression::zstd::*` takes only `ArrayRef<uint8_t>`, so accepting `StringRef` would not be consistent